Ninjathulhu/Ninjathulhu.Game/Entity.cs

```diff
@@ -3,80 +3,75 @@
 
 namespace Ninjathulhu.Game
 {
-    class EntityComponents
+    public class EntityComponents
     {
-        private Entity Entity;
+        private readonly Entity _entity;
 
-        private Dictionary<Type, Component> Components = new Dictionary<Type, Component>();
+        private readonly Dictionary<Type, Component> _components = new Dictionary<Type, Component>();
 
         public EntityComponents(Entity entity)
         {
-            Entity = entity;
+            _entity = entity;
         }
 
         public Component Get<TComponent>()
             where TComponent : Component
         {
-            return Components[typeof(TComponent)];
+            return Get(typeof (TComponent));
```

Comment:
Changed all of these `Foo<TComponent>()` methods to delegate to `Foo(Type)`. Kind of a weird way to manage components, but I think I can get the hang of it.